modules/data-loading/pages/kafka-loader/kafka-ssl-sasl.adoc

```diff
@@ -0,0 +1,218 @@
+* Set up SASL
+* Set up SSL and SASL
+
+== Set up one-way SSL
+You can set up SSL to only have the client (TigerGraph) verify the identity of the server (your external Kafka cluster or broker).
+
+=== Before you begin
+The following prerequisites should be met on the server side:
+
+* Ensure you have generated a Certificate Authority (CA) on the server.
+* Ensure you have generated a certificate and private key with the server's hostname as the alias on the server.
+The certificate has been signed by the CA.
+* Ensure the IP address and the fully qualified domain name (FQDN) of the server is listed in the `/etc/hosts` file on the server.
+
+You will need the following files in order to configure SSL on TigerGraph.
+
+* The truststore file from the broker
+* The broker's Certificate Authority
+
+=== Procedure
+. Copy the following files from the server side to your TigerGraph cluster's m1 machine:
+* The truststore file from the broker
+** For example, `server.truststore.jks`
+* The certificate for the broker's CA
+. xref:kafka-loader/kafka-ssl-sasl.adoc[Create the Kafka data source] and specify SSL configurations in the datasource configuration JSON file.
+For example:
++
+[.wrap,json]
+----
+{
+   "broker": "kafka-0.tigergraph.com:9092", <1>
+   "kafka_config":{
+       "security.protocol": "SSL",
+       "ssl.endpoint.identification.algorithm": "",
+       "ssl.ca.location":"/home/tigergraph/SSL_one_way/ca-root.crt",
+       "ssl.truststore.location":"/home/tigergraph/SSL_one_way/server.truststore.jks", <2>
+       "ssl.truststore.password":"tiger123"
+   }
+}
+----
+<1> The hostname and port of the broker.
+<2> `ssl.ca.location` and `ssl.truststore.location` should match the paths where you copied the certificate authority and the truststore file.
+
+Once you have defined the data source with the correct SSL configuration, you have configured one-way SSL.
+When you run loading jobs from this data source, SSL secures data in transit and the client verifies the identity of the server.
+
+
+== Set up two-way SSL
+
+You can set up two-way SSL between TigerGraph and your external Kafka cluster.
+Two-way authentication means the client authenticates the server and the server also authenticates the client.
+
+=== Before you begin
+* Ensure you have link:https://docs.confluent.io/platform/current/kafka/authentication_ssl.html#brokers[enabled SSL authentication for your Kafka cluster].
+You only need to follow the instructions in Confluent's documentation to configure brokers.
+The steps for configuring the client are described on this page.
+** For optional settings, `ssl.endpoint.identification.algorithm` should be left empty.
+
+* Ensure you have created a certificate for the client, and signed it with the broker's CA.
+* Ensure the IP address and the fully qualified domain name (FQDN) of both the server and the client are in the `/etc/hosts` file of the server.
+
+=== Procedure
+. Copy the following files from the broker to every node:
```

Comment:
After copying all files to all tigergraph nodes, we need to specify the owner of all files as tigergraph.
`chown -R tigergraph:tigergraph <file path>`